Core/src/Propagator/detail/CovarianceEngine.cpp

```diff
@@ -15,6 +15,53 @@
 
 namespace Acts {
 namespace {
+/// @brief Perform a tiled matrix multiplication, avoiding Eigen GEMM methods.
+///
+/// @tparam M First dimension of first matrix
+/// @tparam N Second dimension of first matrix, and first dimension of second
+/// matrix
+/// @tparam P Second dimension of second matrix
+///
+/// @param[in] a MxN matrix
+/// @param[in] b NxP matrix
+///
+/// @returns The product ab
+template <std::size_t M, std::size_t N, std::size_t P>
+inline ActsMatrix<M, P> tiledMult(const ActsMatrix<M, N>& a,
```

Comment:
How useful do you think is is to pull out of this compilation unit into a helper function somewhere, and potentially add a unit test against a regular multiplication (if we think this should be generally correct)?

Reply:
Yeah, that would be a solid idea. Where do you think would be the best place to put this?

Reply:
I'd put it here https://github.com/acts-project/acts/blob/main/Core/include/Acts/Utilities/Helpers.hpp.

Also this could be done in a follow-up PR, up to you.

Reply:
Nah to hell with it I'll do it now. Clean git history and all.

Reply:
Moved to #1184.